source/documentation/other-topics/aws-es-encryption-migration.html.md.erb

```diff
@@ -0,0 +1,201 @@
+
+```hcl
+module "encrypted_es" {
+  source                          = "github.com/ministryofjustice/cloud-platform-terraform-elasticsearch?ref=3.8.0"
+  cluster_name                    = var.cluster_name
+  cluster_state_bucket            = var.cluster_state_bucket
+  application                     = var.application
+  business-unit                   = var.business-unit
+  environment-name                = var.environment-name
+  infrastructure-support          = var.infrastructure-support
+  is-production                   = var.is-production
+  team_name                       = var.team_name
+  elasticsearch-domain            = "encrypted-example-es"
+  encryption_at_rest              = true
+  node_to_node_encryption_enabled = true
+  namespace                       = var.namespace
+  elasticsearch_version           = "7.9"
+  aws-es-proxy-service            = "aws-es-proxy-service-encrypted"
+  aws-es-proxy-replica-count      = 2
+  instance_type                   = "t3.medium.elasticsearch"
+  s3_manual_snapshot_repository   = module.es_snapshots_s3_bucket.bucket_arn
+}
+
+module "ns_annotation" {
+  source              = "github.com/ministryofjustice/cloud-platform-terraform-ns-annotation?ref=0.0.2"
+  ns_annotation_roles = [module.source_es.aws_iam_role_name, module.encrypted_es.aws_iam_role_name]
+  namespace           = var.namespace
+}
+
+
+```
+
+### Step 3: Stop all apps connected to the Elasticsearch cluster
+
+### Step 4: Snapshot source Elasticsearch
+
+  - Port forward the source `aws-es-proxy-service` 
+
+  ```
+  kubectl \
+    -n [your namespace] \
+    port-forward \
+    svc/aws-es-proxy-service 9200:9200
+  ```
+
+  - Register the snapshot repository inside Elasticsearch cluster via the [ES API](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html). 
+    
+    In this guide, same bucket is used with multiple domains. Hence adding a basepath will keeps the snapshots in separate directories.
+
+    A sample template is given below:
+
+  ```
+    PUT http://localhost:9200/_snapshot/<repo-name-for-snapshot>
+  {
+    "type": "s3",
+    "settings": {
+      "bucket": "<s3-bucket for snapshots>",
+      "base_path": "<repo-name for snapshot>",
+      "region": "eu-west-2",
+      "role_arn": "<role arn from es_snapshots_role secret>"
+    }
+  }
+  ```
+
+  - Create a snapshot 
+
+  ```
+  PUT http://localhost:9200/_snapshot/<repo-name-for-snapshot>/latest-snapshot
+  ```
+
+### Step 5: Restore to encrypted Elasticsearch
+
+  - Port forward the encrypted Elasticsearch cluster
+
+  ```
+  kubectl \
+    -n [your namespace] \
+    port-forward \
+    svc/aws-es-proxy-service-encrypted 9200:9200
+  ```
+
+  - Restore snapshot
+
+  NOTE: `"include_global_state":true` will restore all the key data
+
+  ```
+  POST http://localhost:9200/_snapshot/<repo-name-for-snapshot>/<snapshot-you-want-to-restore-from>/_restore
+  {
+    "include_global_state": true
+  }
+  ```
+
+### Step 6: Connect apps to new ES
+
```

Comment:
Maybe mention here that the new apps need to use the new es_proxy service.   Or they can use the terraform outputted proxy URL.